storage/innobase/include/sync0policy.h

```diff
@@ -157,11 +192,11 @@ class MutexDebug {
 	/** Mutex is being destroyed. */
 	void destroy() UNIV_NOTHROW
 	{
-		ut_ad(m_context.m_thread_id == os_thread_id_t(ULINT_UNDEFINED));
+		ut_ad(m_context.get_thread_id() == os_thread_id_t(ULINT_UNDEFINED));
 
 		m_magic_n = 0;
 
-		m_context.m_thread_id = 0;
+		m_context.set_thread_id(0);
```

Comment:
Neither ```get_thread_id()``` nor ```set_thread_id()``` should be needed here, because ```destroy()``` is to be called when concurrent threads are guaranteed not to access this mutex anymore.

Reply:
That is please keep as is.

Reply:
This question still applies. Is it really needed to do atomic access here?

I think if ```destroy()``` is called while another thread updates ```m_thread_id``` should never happen. And it is good to be able to catch it with things like TSAN.